### DISCUSSION TITLE
Is there any better idea to get screenshot tests with system UI?

### DISCUSSION THREAD

**Discussion by Jaehwa-Noh** [android/nowinandroid#1347] [General]
Now in Android project has roborazzi screenshot tests, but this doesn't contain System UI, so that we can't check this issue #1295 .
You can see the screenshots in this pull request #1343.

The problem is when we use another screenshot test library for get the System UI, we must use same device to pass the test.
For example, if I made screenshot tests with Tablet, other contributor must use same Tablet to pass my screenshot tests, which has the System UI. 
I think it is worst situation that everyone must test it using same device.

Does Anybody have good ideas for this situation?

**Comment by SimonMarquis** [comment]
Hi @Jaehwa-Noh, I'm not sure to fully understand the issue. SnackBars are not part of the SystemUi (unlike Toasts) and therefore should be rendered by Compose without any issue.

**Reply by Jaehwa-Noh** [reply]
Yes, SnackBar isn't System UI, but the problem is that screenshot test without System UI, we can't verify snackbar is whether showing behind system bottom navigation(especially, foldable and Tablet's) or above it, when we just only see the screenshot test.

Is this topic out of screenshot tests?

**Reply by alexvanyo** [reply]
One solution would be to run screenshot tests on device, and then take a screenshot of the device's screen itself with both system UI and app UI.

For a solution fully in app-side code, it should be possible to "fake" insets in some way, either with the Compose internals directly, or the platform insets, and dispatch fake insets locally within a test. Maybe that can all be wrapped up into a custom `DeviceConfigurationOverride`?

Then in the test itself, you could probably "fake" system UI by putting an opaque bar where the fake insets are.

Then, the screenshots could reveal if UI is handling insets properly or if content would be obscured behind the system UI.

**Reply by Jaehwa-Noh** [reply]
Thank you for giving new approach to test.
I would consider the way that makes ```fake``` insets from ```DeviceConfigurationOverride```.

**Reply by Jaehwa-Noh** [reply]
@alexvanyo In this [documentation](https://developer.android.com/reference/kotlin/androidx/compose/ui/test/DeviceConfigurationOverride.Companion), there's no API's for override WindowInsets in ```DeviceConfigurationOverride```.
Do you have any ideas how to give Fake insets?
#1296  My pull request use the safe drawing insets' height to avoid System UI.
```kotlin
val bottomNavigationHeight = density.run {
        WindowInsets.safeDrawing
            .getBottom(density = density)
            .toDp()
    }
```

**Reply by alexvanyo** [reply]
I created a prototype for this in #1498, with an upstream feature request at https://issuetracker.google.com/343474632

**Reply by Jaehwa-Noh** [reply]
Thank you. I'll see it as soon as possible.

**Comment by Jaehwa-Noh** [comment]
I added the parameter for fake height at ```NiaApp```
```kotlin
internal fun NiaApp(
    appState: NiaAppState,
    snackbarHostState: SnackbarHostState,
    showSettingsDialog: Boolean,
    onSettingsDismissed: () -> Unit,
    onTopAppBarActionClick: () -> Unit,
    bottomSystemUiHeight: Dp,
    modifier: Modifier = Modifier,
)
```

**Reply by Phatphatnguyen** [reply]
I want to be
